parsec/mca/pins/pins.c

```diff
@@ -32,6 +34,18 @@ void parsec_pins_instrument(struct parsec_execution_stream_s* es,
                             parsec_task_t* task)
 {
     assert( method_flag < PARSEC_PINS_FLAG_COUNT );
+    /**
```

Comment:
This ties PINS with the profiling system in a new way: if profiling is enabled at configure time, only tasks with a profiling key will instrument that PINS event, if profiling is disabled at configure time, all tasks will instrument that PINS event.

Initially, PINS could be independent from the profiling system. Some PINS instruments (e.g. task_profiler) would do things only if profiling was enabled, but others (e.g. `print_steals` or `iterator_checkers`) would do other kind of printing/checking that does not depend on the profiling.

PRO of this approach: we don't even record the callback function, so overheads are 0 for tasks that are no-profile

CON of this approach: we lose the feature of doing other things than profiling with PINS.

I would advocate fixing the issue (error when no-profile tasks are used) without doing this, unless we also decide to drop the other PINS modules?